### ISSUE TITLE
Cannot load json-schema. The property '@type' has an invalid value 'ceterms:BachelorDegree'

### ISSUE THREAD

**Issue by cwd-mparsons** [CredentialEngine/CredentialRegistry#24]
I am getting the following error on attempting to publish a credential:
{"Result":"{\"errors\":[\"Cannot load json-schema. The property '@type' has an invalid value 'ceterms:BachelorDegree'\"]}","Id":371,"Exception":null,"Status":5,"IsCanceled":false,"IsCompleted":true,"CreationOptions":0,"AsyncState":null,"IsFaulted":false}

Nate had previously communicated an issue with the use of @type, but we thought it was resolved. 

As well, I had included the url parameter of:
       **skip_validation=true**
I thought the latter would allow a publish without validation. Perhaps either the format is incorrect, or the error occurs outside the scope of 'skip_validation'

The credential on our test server is:
http://ctitest.siuccwd.com/credential/83/Bachelor_of_Science_in_Information_Technology_with_a_IT_Generalist_Emphasis

I have attached the json payload, and encoded envelope.

[cred_83_envelope.json.txt](https://github.com/CredentialEngine/CredentialRegistry/files/851932/cred_83_envelope.json.txt)
[cred_83_payload.json.txt](https://github.com/CredentialEngine/CredentialRegistry/files/851933/cred_83_payload.json.txt)

Also note that on any detail page, you can view the generated JSON by clicking the Registry Information panel on the right side of the page:
![image](https://cloud.githubusercontent.com/assets/26332112/24065508/8d58e9ee-0b39-11e7-9efe-ae1d7c0fa86a.png)

**Comment by andersoncardoso** [comment]
Hi, 

The previous problem with `@type` was because the specific classes ('Badge', 'Certificate', etc) were not configured on the app, it only knew 'Credential'. This is fixed. 
But this new issue is coming from the prefix portion, i.e:
The app uses `@type` to find the right schema to validate against, but the current config types are still defined using the `ctdl` prefix. 

Until we update the config with the final schemas and etc, I'm doing a quick fix (adding the 'ceterms' prefixed types on the `config.json`). 
It should be deployed in a few minutes.

**Comment by leolara** [comment]
@andersoncardoso 
I think that either the type that selects the schema should be in the envelop. And not use the one in the actual object.

Where is the commit/Pr for that config.json change?

**Comment by cwd-mparsons** [comment]
@leolara @andersoncardoso 
The recent update enables publishing a credential with validation turned off (skip_validation=true).
I do get errors with validation enabled. 
However, I created a new issue for the latter, as it appears to be a different sources. 
I believe this issue can be closed, dependent on Leo's previous question (Where is the commit/Pr for that config.json change?)

**Comment by andersoncardoso** [comment]
The commit for adding the `ceterms` types to the config is this one: https://github.com/CredentialEngine/CredentialRegistry/commit/cd644454c8961d66bada6501e5448e1b5d23e5e0

keep in mind that this only accepts the `ceterms:<type>` for mapping to the json-schema. But the schema itself it's still the old one, so unless you pass `skip_validation=true` you will receive several errors.

@leolara  This information is about the resource, not the envelope. If we change this, we will have to refactor the validation process, and the 'problem' which is having to map the right type names to the corresponding schemas, will still be there.
Maybe I'm missing the point, but doesn't seem an issue, given that the config is outdated and don't correspond to the data we are passing. We just have to update the config.

**Comment by leolara** [comment]
@andersoncardoso 

 + I thought the envelope had a field with type and that one was used to select the schema, if there is a type in the envelop we should at least make sure it matches the one in the resource
 + If the schema is the old schema that is a huge problem, we should update it ASAP to have the new schema

**Comment by andersoncardoso** [comment]
- we have, but that's the type of the envelope (`envelope_type`), it's more general, a step before validating the resources. The current valid envelop_types are `resource_data`, `paradata` or `json_schema`.  
Changing that would require a big refactor (~6h to 8h work, since besides the code, all tests, docs and samples would need change).
if it's decided that this should be done, add an issue and assign me :+1:

- I worked on this just now, but had some problems, I explained better here: https://github.com/CredentialEngine/CredentialRegistry/issues/26
